### PR TITLE
Integration testing for group management

### PR DESCRIPTION
**Pull request by tanmoysrt**
#### Describe the changes you have made in this PR -
Integration tests for group management
  - delete group
  - add a secondary mentor
  - remove the secondary mentor
  - convert member to mentor
  - convert mentor to a member

Note: Please check **Allow edits from maintainers.** if you would like us to assist in the PR.

### CODE REVIEW

File: spec/system/group_spec.rb [hunk 1]
```diff
@@ -67,4 +63,66 @@
...
+    click_on "Remove"
+    execute_script "document.getElementById('deletememberModal').style.display='block'"
+    execute_script "document.getElementById('deletememberModal').style.opacity=1"
+    click_on "Delete"
+
+    expect(page).to have_text("Group member was successfully removed.")
+  end
+
+  it "convert member to mentor" do
+    GroupMember.create(group_id: @group.id, user_id: @user2.id, mentor: false)
```
Comment (smritigarg) [inline spec/system/group_spec.rb:100]: Can we move this line to a `let` statement to avoid code repetition?

File: spec/system/group_spec.rb [hunk 2]
```diff
@@ -67,4 +63,66 @@
...
+    visit "/groups/#{@group.id}"
```
Comment (smritigarg) [inline spec/system/group_spec.rb:101]: This can be moved to a common subject 🤔 ?

File: spec/system/group_spec.rb [hunk 3]
```diff
@@ -67,4 +63,66 @@
...
+    make_mentor_btn.click
+    execute_script "document.getElementById('promote-member-modal').style.display='block'"
+    execute_script "document.getElementById('promote-member-modal').style.opacity=1"
+    make_mentor_btn = find(id: "groups-member-promote-button")
+    make_mentor_btn.click
+
+    expect(page).to have_text("Group member was successfully updated.")
+  end
+
+  it "convert mentor to member" do
```
Comment (smritigarg) [inline spec/system/group_spec.rb:112]:
```suggestion
  it "converts mentor to member" do
```

File: spec/system/group_spec.rb [hunk 4]
```diff
@@ -67,4 +63,66 @@
...
+  it "deletes the group" do
+    visit user_groups_path(@user)
+    click_on "Delete"
+    button = find(id: "groups-group-delete-button")
+    button.click
+
+    expect(page).to have_text("Group was successfully deleted.")
+  end
+
+  it "add secondary mentor" do
```
Comment (smritigarg) [inline spec/system/group_spec.rb:76]:
```suggestion
  it "adds secondary mentor" do
```

File: spec/system/group_spec.rb [hunk 5]
```diff
@@ -67,4 +63,66 @@
...
+    fill_in_input "#group_email_input_mentor", with: @user3.email
+    fill_in_input "#group_email_input_mentor", with: :enter
+    click_button "Add mentors"
+
+    expect(page).to have_text(
+      "Out of 1 Email(s), 1 was valid and 0 were invalid. 1 user(s) will be invited."
+    )
+  end
+
+  it "remove secondary mentor" do
```
Comment (smritigarg) [inline spec/system/group_spec.rb:88]:
```suggestion
  it "removes secondary mentor" do
```

File: spec/system/group_spec.rb [hunk 6]
```diff
@@ -67,4 +63,66 @@
...
+    visit "/groups/#{@group.id}"
+    click_on "Remove"
+    execute_script "document.getElementById('deletememberModal').style.display='block'"
+    execute_script "document.getElementById('deletememberModal').style.opacity=1"
+    click_on "Delete"
+
+    expect(page).to have_text("Group member was successfully removed.")
+  end
+
+  it "convert member to mentor" do
```
Comment (smritigarg) [inline spec/system/group_spec.rb:99]:
```suggestion
  it "converts member to mentor" do
```

**Review by smritigarg — COMMENTED** [review]
Good work @tanmoysrt Just some refactoring suggestions for avoiding code repetition, 
otherwise looks fantastic 👏

File: spec/system/group_spec.rb (hunk 1 above)
Comment (tanmoysrt) [reply spec/system/group_spec.rb:100]: yes we can do that

**Comment by tanmoysrt** [comment]
@smritigarg updated the PR accordingly the suggestion. Please check out

**Review by smritigarg — APPROVED** [review]
Thanks for the update @tanmoysrt , Approving the changes

File: spec/system/group_spec.rb [hunk 7]
```diff
@@ -60,11 +59,75 @@
...
+  it "deletes the group" do
+    visit user_groups_path(primary_mentor)
+    click_on "Delete"
+    button = find(id: "groups-group-delete-button")
+    button.click
+
+    expect(page).to have_text("Group was successfully deleted.")
+  end
+
+  it "adds secondary mentor" do
```
Comment (vedant-jain03) [inline spec/system/group_spec.rb:79]:
```suggestion
  it "adds mentor" do
```

File: spec/system/group_spec.rb [hunk 8]
```diff
@@ -60,11 +59,75 @@
...
+    fill_in_input "#group_email_input_mentor", with: user.email
+    fill_in_input "#group_email_input_mentor", with: :enter
+    click_button "Add mentors"
+
+    expect(page).to have_text(
+      "Out of 1 Email(s), 1 was valid and 0 were invalid. 1 user(s) will be invited."
+    )
+  end
+
+  it "removes secondary mentor" do
```
Comment (vedant-jain03) [inline spec/system/group_spec.rb:91]:
```suggestion
  it "removes mentor" do
```

**Review by vedant-jain03 — APPROVED** [review]

**Pull request merged**